File: CountLines/CountLines1.c
#include <stdio.h>

// http://stackoverflow.com/a/12733630/5086987
int countlines( char *filename ) {
  int lines = 0;

  FILE *file = fopen( filename, "r" );
  if ( file != NULL ) {
    int character = 0;
    while ( ! feof( file ) ) {
      character = fgetc( file );
      if ( character == '\n' ) {
        lines++;
      }
    }

    fclose( file );
  }

  return lines;
}

int main( int argc, char *argv[] ) {
  printf( "%d\n", countlines( argv[ 1 ] ) );
}

File: CountLines/CountLines1.go
package main

import (
  "fmt"
  "os"
  "bufio"
)

const
  BUFFER_SIZE = 8 * 1024

// http://stackoverflow.com/a/24563853/5086987
func CountLines( filename string ) int {
  lines := 0

  file, err := os.Open( filename )
  if err == nil {
    reader  := bufio.NewReader( file )
    scanner := bufio.NewScanner( reader )

    defer file.Close()

    for scanner.Scan() {
      lines += 1
    }
  }

  return lines
}

func main() {
  fmt.Println( CountLines( os.Args[ 1 ] ) )
}
